### PR TITLE
Endpoint/handle db range query

### PR DESCRIPTION
**Pull request by Flaiizor**
So, what happened here is I created a new endpoint not realising that there already was an endpoint that did the thing. So I removed the old endpoint and made a new one, now with better functionality and more love

### CODE REVIEW

File: src/boost-api/Endpoint_Handlers.cpp [hunk 1]
```diff
@@ -76,52 +76,61 @@ namespace api {
...
-            Trajectory_Manager::db_range_query(table, window);
-        } catch (const std::exception &e) {
+            // Determine the db_table enum based on the table_key
+            db_table db_table{};
+            if (table_key == "original") {db_table = db_table::original_trajectories; }
+            else if (table_key == "simplified") {db_table = db_table::simplified_trajectories; }
+            else { throw std::runtime_error("Error in db_table, must be either 'simplified' or 'original' "); }
+
+            // Extract window object or use defaults
+            spatial_queries::Range_Query::Window window;
```
Comment (Sarmisuper) [inline src/boost-api/Endpoint_Handlers.cpp]: initialize

File: src/boost-api/Endpoint_Handlers.cpp [hunk 2]
```diff
@@ -76,52 +76,61 @@ namespace api {
...
+            const boost::json::object &windowObj = jsonObject.contains("window") ?
+                                                   jsonObject.at("window").as_object() :
+                                                   boost::json::object(); // Empty object if not present
+
+            // Assign values from JSON or use defaults
+            window.x_low = windowObj.contains("x_low") ? windowObj.at("x_low").as_double() : std::numeric_limits<double>::lowest();
```
Comment (Sarmisuper) [inline src/boost-api/Endpoint_Handlers.cpp]: hmm, du bruger lowest. Hvad er forskellen mellem lowest og min?

File: src/boost-api/Endpoint_Handlers.cpp [hunk 3]
```diff
@@ -76,52 +76,61 @@ namespace api {
...
+            window.x_high = windowObj.contains("x_high") ? windowObj.at("x_high").as_double() : std::numeric_limits<double>::max();
+            window.y_low = windowObj.contains("y_low") ? windowObj.at("y_low").as_double() : std::numeric_limits<double>::lowest();
+            window.y_high = windowObj.contains("y_high") ? windowObj.at("y_high").as_double() : std::numeric_limits<double>::max();
+            window.t_low = windowObj.contains("t_low") ? windowObj.at("t_low").as_int64() : std::numeric_limits<long>::lowest();
+            window.t_high = windowObj.contains("t_high") ? windowObj.at("t_high").as_int64() : std::numeric_limits<long>::max();
```
Comment (Sarmisuper) [inline src/boost-api/Endpoint_Handlers.cpp]: unsigned long i numeric limits

File: src/querying/Range_Query.hpp [hunk 4]
```diff
@@ -3,17 +3,18 @@
...
-            double y_low{};
-            double y_high{};
-            unsigned long t_low{};
-            unsigned long t_high{};
+            double x_low{ std::numeric_limits<double>::lowest() };
+            double x_high{ std::numeric_limits<double>::max() };
+            double y_low{ std::numeric_limits<double>::lowest() };
+            double y_high{ std::numeric_limits<double>::max() };
+            long t_low{ std::numeric_limits<unsigned long>::lowest() };
+            long t_high{ std::numeric_limits<unsigned long>::max() };
```
Comment (Sarmisuper) [inline src/querying/Range_Query.hpp]: ændre til unsigned

File: src/boost-api/Endpoint_Handlers.cpp (hunk 3 above)
Comment (Sarmisuper) [inline src/boost-api/Endpoint_Handlers.cpp]: Du behøver vel ikke at sætte dem til min og max hvis du har sat deres default værdier i range_query.hpp

**Review by Sarmisuper — CHANGES_REQUESTED** [review]

File: src/boost-api/Endpoint_Handlers.cpp (hunk 1 above)
Comment (Flaiizor) [reply src/boost-api/Endpoint_Handlers.cpp]: initialized

File: src/boost-api/Endpoint_Handlers.cpp (hunk 2 above)
Comment (Flaiizor) [reply src/boost-api/Endpoint_Handlers.cpp]: Difference? no, just min now

File: src/boost-api/Endpoint_Handlers.cpp (hunk 3 above)
Comment (Flaiizor) [reply src/boost-api/Endpoint_Handlers.cpp]: yes

File: src/querying/Range_Query.hpp (hunk 4 above)
Comment (Flaiizor) [reply src/querying/Range_Query.hpp]: yes

File: src/boost-api/Endpoint_Handlers.cpp (hunk 3 above)
Comment (Flaiizor) [reply src/boost-api/Endpoint_Handlers.cpp]: Nej, det er sandt

**Review by Sarmisuper — APPROVED** [review]

**Pull request merged**